ADR/0032-decoupling-deployment.md

```diff
@@ -0,0 +1,201 @@
+  contributor) has to adopt the whole thing, or nothing.
+- Today, we can generate simple but often incorrect deployment manifests for users' applications,
+  which we deploy via the application gitops repo. The user cannot interact with or influence these
+  resources directly; they have to do so through the Application/Component API. If we permit them to
+  directly provide deployment resources through that API, the API will become a leaky abstraction.
+  If don't permit them to provide their own resources, and instead try to handle every application
+  configuration case in the HAS generation code, we will struggle to keep up with users' demands for
+  the many and varied kinds of application deployments.
+
+See also [RHTAP-1873](https://issues.redhat.com/browse/RHTAP-1873).
+
+## Decision
+
+We are going to decouple the deployment from the build, test, and release portions of our system.
+
+**Deployment**:
+
+- The [Environment], [SnapshotEnvironmentBinding], and [GitOpsDeploymentManagedEnvironment]
+  resources will be deprecated and eventually decomissioned.
+- The [application-service] will stop generating GitOps repo content.
+- HAC will stop rendering [Environments] and their status.
+- If a user wants to make use of deployment capabilities, we will promote the usage of [renovatebot]
+  to propagate _released_ images to their self-managed gitops repo as pull requests.
+
+**Test and promotion**:
+
+- We will retain the [DeploymentTargetClaim], [DeploymentTarget], and [DeploymentTargetClass] APIs.
+  Collectively, call these the "DTC" APIs. These should have no dependence on other AppStudio APIs.
+  Only [integration-service] should depend on them. The controllers that support the DTC API should
+  be organized into their own project: "the [deployment-target-operator]", or something like that.
+- [integration-service] should be modified to create and destroy [DeploymentTargetClaims] directly
+  instead of [Environment] resources. Users will be expected to provide integration test pipelines
+  that accept a kubeconfig and snapshot as they do now, but those pipelines should now take initial
+  steps to *deploy the instance of their application to be tested* to the deployment target provided
+  by [integration-service] using the provided kubeconfig.
+- We should promote [release-service] as the primary means to advertise to [renovatebot] that one or
+  more images have passed testing and are ready to be promoted to a particular environment (with
+  a lowercase "e") by way of image tags in a registry.
+
+### Out of scope
+
+Some other interesting ideas that are floating around these days, but which should be taken up in
+other ADRs, if we take them up at all:
+
+- Decoupling [integration-service] and [release-service] from a shared [Snapshot] API. They will
+  continue to share a [Snapshot] API as of this ADR.
+- Decoupling [integration-service] from the [application-service] APIs. It will still promote images
```

Comment:
Can you refer to the APIs specifically like you did with Snapshot? I assume that the primary CRD is the Component.

Reply:
Done in bf4520b.